handler/env.go

```diff
@@ -0,0 +1,29 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/sethvargo/go-envconfig"
+)
+
+// exportEnvVar ensures we export resolved values back into the environment
+// This is useful for cases where we set a default in our env struct, and need
+// the value to be propagated to other processes.
+func exportEnvVar(_ context.Context, _, resolvedKey, _, resolvedValue string) (newValue string, stop bool, err error) {
+	os.Setenv(resolvedKey, resolvedValue)
+	return resolvedValue, false, nil
+}
+
+// ProcessEnv populates struct from environment variables.
+func ProcessEnv(ctx context.Context, v any) error {
+	err := envconfig.ProcessWith(ctx, &envconfig.Config{
+		Target:   v,
+		Mutators: []envconfig.Mutator{envconfig.MutatorFunc(exportEnvVar)},
```

Comment:
so this'll end up taking all the resolved config values and re-exporting them all back to env?

Reply:
Yes. Unfortunately there's no way to tell if an environment variable has changed in the process. I was initially hopeful that I could compare `originalValue` vs `resolvedValue`, but in practice `envconfig` sets the default early enough that we can't disambiguate between the cases.